### ISSUE TITLE
Splicing reader conditionals in `:require` form triggers `:unsorted-required-namespaces`

### ISSUE THREAD

**Issue by dcostaras** [clj-kondo/clj-kondo#1571]
**version**

```
❯ clj-kondo --version
clj-kondo v2022.01.15
```

**platform**

macOS, native clj-kondo from brew (`brew "borkdude/brew/clj-kondo"`)

**problem**

The `:unsorted-required-namespaces` linter is being triggered when using splicing reader conditionals

**repro**

https://github.com/dcostaras/clj-kondo-repro

**config**

https://github.com/dcostaras/clj-kondo-repro/blob/main/.clj-kondo/config.edn

**expected behavior**

We would like to turn on that linter for our codebase. Our convention is to have all universal requires at the top of the :require form and then one splicing reader conditional last. 

Ideally we'd like it to check the sorting of each section, however at the least to not throw an error.

We'd be interested contributing a PR if it wasn't too onerous.
